### PR TITLE
Widget improvements/fixes

### PR DESCRIPTION
**Pull request by midwan**
- Use different color when a widget is Enabled or Disabled
- Fixed: bug with button widgets triggering by mouse events even if mouse wasn't on top of them
- Fixed: some widgets didn't have enough spacing or had hardcoded values
- Fixed: Listbox widget would iterate over all elements, even if they were not visible on screen. Now using CurrentClipArea instead.

### CODE REVIEW

File: src/widgets/button.cpp [hunk 1]
```diff
@@ -166,7 +166,10 @@ namespace gcn
         graphics->drawLine(getWidth() - 1, 1, getWidth() - 1, getHeight() - 1);
         graphics->drawLine(1, getHeight() - 1, getWidth() - 1, getHeight() - 1);
 
+		if (isEnabled())
```
Comment (kallisti5) [inline src/widgets/button.cpp:169]: spacing is weird

File: src/widgets/button.cpp [hunk 2]
```diff
@@ -238,13 +241,10 @@ namespace gcn
     {
         if (mMousePressed)
         {
-            return mHasMouse;
+			return true;
```
Comment (kallisti5) [inline src/widgets/button.cpp:244]: more weird spacing

File: src/widgets/checkbox.cpp [hunk 3]
```diff
@@ -94,7 +94,10 @@ namespace gcn
         drawBox(graphics);
 
         graphics->setFont(getFont());
+		if (isEnabled())
```
Comment (kallisti5) [inline src/widgets/checkbox.cpp:97]: looks like tabs vs spaces?

**Review by kallisti5 — CHANGES_REQUESTED** [review]
lots of spacing issues.  Overall the changes look good however.   If you can clean up the tab vs space i'll merge and draft a new release :-)

File: src/widgets/button.cpp (hunk 1 above)
Comment (midwan) [reply src/widgets/button.cpp]: Fixed with 25c670187742a8c0dbe560b120b257c5bc93993f

File: src/widgets/button.cpp (hunk 2 above)
Comment (midwan) [reply src/widgets/button.cpp]: Fixed with 25c670187742a8c0dbe560b120b257c5bc93993f

File: src/widgets/checkbox.cpp (hunk 3 above)
Comment (midwan) [reply src/widgets/checkbox.cpp]: Fixed with 25c670187742a8c0dbe560b120b257c5bc93993f

**Comment by midwan** [comment]
Sorry about the spacing issues, I missed those - I have more changes in my own version, and I was trying to bring them over one step at a time. I guess the code comparison tool I used didn't care much about retaining the spacing :)

**Pull request merged**